src/modules/dashboard/sagas/extensions.ts

```diff
@@ -165,18 +167,43 @@ function* colonyExtensionEnable({
       throw new Error('Extension not installed');
     }
 
+    /*
+     * Upload whitelist policy to IPFS
+     */
+    let agreementHash = '';
+    if (
+      extensionId === Extension.Whitelist &&
```

Comment:
So I'm not particularly happy with this, but right now it's the best we can do.

However, if we'll have to do this for more extensions, we'll need to think about refactoring this saga.

The way I'm thinking about this to work in the future, is something like each extension could have it's own individual _enable_ saga _(same saga for installation, just the enable saga to be different)_

But we can see about that, if we get to that point.